### PR TITLE
refactor: Remove old safe creation code

### LINKED ISSUE #1404: Remove old safe creation and AB test
## Description

The A/B test results for the new safe creation flow were positive so we want to clean up the old components and the A/B test setup.

## Requirements

- Copy existing tests from `src/components/create-safe` into the new directory `src/components/new-safe`
- Remove `src/components/create-safe`
- Remove AB test related logic (this will most likely be replaced at a later time with something like Google Optimize)

### PR DESCRIPTION
**Pull request by usame-algan**
## What it solves

Resolves #1404 

## How this PR fixes it

- Removes old safe creation components under `/src/components/create-safe`
- Moves applicable tests to new directory `/src/components/new-safe/create`
- Removes abTest for safe creation
- Renames create safe steps to be more descriptive

## How to test it

1. Open the Safe
2. Go through the safe creation flow
3. Observe that it still works

### CODE REVIEW

**Review by katspaugh — CHANGES_REQUESTED** [review]
We shouldn't remove the A/B test feature from GTM.
It was meant to be reusable.

**Comment by usame-algan** [comment]
> We shouldn't remove the A/B test feature from GTM.

I restored the ab test related code and only removed its initialization from the app

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -51,7 +49,6 @@ const InitApp = (): null => {
   useTxPendingStatuses()
   useTxTracking()
   useBeamer()
-  useABTesting(AbTest.SAFE_CREATION)
```
Comment (katspaugh) [inline src/pages/_app.tsx:54]: Please also remove `SAFE_CREATION` from the enum.

**Review by katspaugh — APPROVED** [review]
Looks good, just one last comment.

File: src/pages/_app.tsx (hunk 1 above)
Comment (usame-algan) [reply src/pages/_app.tsx:54]: I've removed it and added a comment so we know where to put future abTest entries

**Pull request merged**